sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -376,6 +400,170 @@ private static MessagingEntityType validateEntityPaths(ClientLogger logger, Stri
         return entityType;
     }
 
+
+    @ServiceClientBuilder(serviceClients = {ServiceBusSessionReceiverAsyncClient.class})
+    public final class ServiceBusSessionReceiverAsyncClientBuilder {
+        private String topicName;
+        private String queueName;
+        private String subscriptionName;
+        private ReceiveMode receiveMode;
+        private String sessionId;
+        ServiceBusSessionReceiverAsyncClientBuilder () {
+
+        }
+        /**
+         * Return instance of {@link ServiceBusMultiSessionProcessorClientBuilder} to receive messages from multiple sessions.
+         * @return {@link ServiceBusSessionReceiverAsyncClient} instance.
+         */
+        public ServiceBusSessionReceiverAsyncClient buildAsyncClient() {
+            return null;
+        }
+        public ServiceBusSessionReceiverAsyncClientBuilder receiveMode(ReceiveMode receiveMode) {
+            this.receiveMode = receiveMode;
+            return this;
+        }
+        public ServiceBusSessionReceiverAsyncClientBuilder sessionId(String sessionId) {
```

Comment:
null would be acceptable here? And then it's just "get next available session" behavior?